src/gasEstimator/gasEstimator.ts

```diff
@@ -0,0 +1,50 @@
+import type { BigNumber } from 'ethers';
+import { isDeployRequest } from '../common/relayRequest.utils';
+import type { EnvelopingTxRequest } from '../common/relayTransaction.types';
+import RelayClient from '../RelayClient';
+import {
+  standardMaxPossibleGasEstimation,
+  linearFitMaxPossibleGasEstimation,
+} from './utils';
+
+const estimateRelayMaxPossibleGas = async (
+  request: EnvelopingTxRequest,
+  relayWorkerAddress: string
+): Promise<BigNumber> => {
+  const {
+    relayRequest,
+    metadata: { signature },
+  } = request;
+
+  const { tokenContract, tokenAmount } = relayRequest.request;
+  const { feesReceiver, callForwarder, gasPrice } = relayRequest.relayData;
+
+  const relayClient = new RelayClient();
```

Comment:
By doing this here I think that we are creating a dependency on the relayClient and the constructor change, we will need to include those changes here or even pass the client as parameter.

Reply:
Does this mean we're going to have a circular dependency?

Reply:
not circular but we are coupling this method with the relayClient.